pylint/checkers/exceptions.py

```diff
@@ -254,6 +255,15 @@ class ExceptionsChecker(checkers.BaseChecker):
 
     def open(self) -> None:
         self._builtin_exceptions = _builtin_exceptions()
+        if any(
+            "." in exc_name for exc_name in self.linter.config.overgeneral_exceptions
+        ):
+            warnings.warn(
+                "Specifying exception names in overgeneral-exceptions option without"
+                " module name has been deprecated."
+                " Use fully qualified name (e.g. builtins.Exception) instead.",
+                DeprecationWarning,
```

Comment:
I have recently learned about `stacklevel`: https://github.com/PyCQA/pylint/pull/7465

We should probably use it here as well.

Reply:
As explained in my other review comment about deprecation warnings, Python's warnings module is probably not the right tool here. But to address this, regardless of `stacklevel` we choose, it will look weird to the user as it will point at pylint's code (just at a different point in call stack):
```
/home/ubuntu/work/pylint/pylint/checkers/exceptions.py:259: DeprecationWarning: Specifying exception names in overgeneral-exceptions option without module name has been deprecated. Use fully qualified name (e.g. builtins.Exception) instead.
  warnings.warn(
```

Reply:
My point was that the calling code is going to be pylint regardless of how far up the stack we go (well ok, technically if you go high enough, you'll probably see stdlib's runpy module instead :)) as this deprecation warning is supposed to be shown when you run `pylint` cli - it's not about some plugin API in pylint.
I talked about it a bit more in the other review comment thread:
https://github.com/PyCQA/pylint/pull/7497#discussion_r975516077

Reply:
Ha right. It's going to be fired when the old value without qualified name is used in a conf file or in CLI. Shouldn't it be a UserWarning instead ?